pkg/controller/oneagent/oneagent_controller.go

```diff
@@ -384,13 +384,18 @@ func (r *ReconcileOneAgent) updateCR(instance *dynatracev1alpha1.OneAgent) error
 
 func newDaemonSetForCR(instance *dynatracev1alpha1.OneAgent) *appsv1.DaemonSet {
 	selector := buildLabels(instance.Name)
+	if instance.Labels != nil && len(instance.Labels) > 0 {
+		for k, v := range instance.Labels {
+			selector[k] = v
+		}
+	}
+
 	podSpec := newPodSpecForCR(instance)
 
 	return &appsv1.DaemonSet{
 		ObjectMeta: metav1.ObjectMeta{
 			Name:      instance.Name,
 			Namespace: instance.Namespace,
-			Labels:    selector,
```

Comment:
Do we want to not have any labels for the DS?

Reply:
This is not allowed, somehow. Throws an error ; refer - https://circleci.com/gh/Dynatrace/dynatrace-oneagent-operator/1070?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link

Reply:
```
pkg/controller/oneagent/oneagent_controller.go:400: cannot use promoted field ObjectMeta.Labels in struct literal of type "k8s.io/api/core/v1".PodTemplateSpec
```

This was for this line: https://github.com/Dynatrace/dynatrace-oneagent-operator/pull/183/commits/4bb823971fab36e3e1355da6fa40bde23c19457d#diff-a0ddcc1928501dbec12491bace283fa3R400 which was not the 393 mentioned above.